### PR TITLE
Code documentation

### LINKED ISSUE #11: Add code documentation
Adding docstrings to the modules will surely help in better understanding of what is going on inside of it.
We have a lot of undocumented code in coyote module as well as in our js files.

For python, follow the format:
```python
def function(param1):
    """
    Parameters:
    param1 [type] => its description
    --------------------------------------------------
    Returns:
    output1 [type] => its description
    """

    # Logic..

    return(output1)
```

For javascript:
```javascript
function name(param1) {
    /*
    @param {Type} param1 - description
    @returns {Type} output1 - description
    */

    // Logic..
    
   return output
}
```

### PR DESCRIPTION
**Pull request by 142ayushkumar**
Closes #11

### CODE REVIEW

**Review by kaustubhhiware — COMMENTED** [review]
Great work @142ayushkumar :smile: 

Just need to make some grammatical / spelling corrections, then we're off.
I'll also request you to try to make it a bit more non-technical.
I'll wait for @Demfier and @kingofools to chip in as well, but I'll suggest the basic corrections.

File: coyote/views.py [hunk 1]
```diff
@@ -24,6 +24,11 @@ def get_domain(url):
 
 
 def home(request):
+    """
+    it takes necessary data from frontend request and sent it to hackerearth API
+    if the code doesnt compiles, then it takes necessary keyword from error messages
```
Comment (kaustubhhiware) [inline coyote/views.py]: `if the code doesnt compiles` can be converted to `if the code doesn't compile`

`then it takes the necessary keyword` => `then it finds the necessary keyword`

File: coyote/views.py [hunk 2]
```diff
@@ -24,6 +24,11 @@ def get_domain(url):
 
 
 def home(request):
+    """
+    it takes necessary data from frontend request and sent it to hackerearth API
```
Comment (kaustubhhiware) [inline coyote/views.py]: `it takes necessary data` => `it takes the inputted code`
See, for a newcomer, the explanation provided must be clear. The explanation seems to be written by someone who did not give enough time to read the function.

`sent it to hackerearth API` => `sends it to hackerearth API` (although technically incorrect, still acceptable)

File: coyote/views.py [hunk 3]
```diff
@@ -24,6 +24,11 @@ def get_domain(url):
...
+    if the code doesnt compiles, then it takes necessary keyword from error messages 
+    and google the result with some regrex matching and filtering to get required debug links
```
Comment (kaustubhhiware) [inline coyote/views.py]: * and searches for it on google with regex (spelled regrex) matching 
* remove _some_
* required => helpful or _get required_ => suggest

Comment (kaustubhhiware) [reply coyote/views.py]: The documentation cannot contain words like some, required. Documentation must be deterministic, leaving nothing to chance.

File: coyote/views.py [hunk 4]
```diff
@@ -87,4 +92,7 @@ def home(request):
 
 
 def codeplay(request):
+    """
+    Given request return HttpResponse with that rendered text
```
Comment (kaustubhhiware) [inline coyote/views.py]: Please take some time to get an understanding of Django.
The function does not "get request" per say. 

The function re-renders the htmlviewer  / codeplay page when this function is called. Please convert to an appropriate format.

**Comment by kaustubhhiware** [comment]
Please do not consider that your work was incorrect because so many suggestions were made. Most of the work is improving upon the original idea after suggestions are made.

Cheer up and take feedback positively :smiley:

**Comment by kaustubhhiware** [comment]
That was quick !
I'll let others also review and approve this, before merging in.

File: coyote/views.py [hunk 5]
```diff
@@ -24,6 +24,11 @@ def get_domain(url):
 
 
 def home(request):
+    """
+    it takes the inputted code from frontend request and sends it to hackerearth API
```
Comment (Demfier) [inline coyote/views.py]: Just add `Parameters` and `Returns` too. Refer #11 for the format. Rest all good.
Nice job @142ayushkumar!

File: coyote/views.py [hunk 6]
```diff
@@ -17,13 +17,32 @@
...
+    Parameters:
+    request[HttpRequest] 
+    --------------------------------------------
+    Returns:
+    render(request, 'init.html') [HttpResponse] => init.html is returned as HttpResponse
+    """
+
+    #it takes the inputted code from frontend request and sends it to hackerearth API
+    #if the code doesn't compile, then it finds the necessary keyword from error messages 
+    #and searches for it on google with regex matching and suggests debug links
```
Comment (Demfier) [inline coyote/views.py]: Moves these description too inside docstring and I guess we can merge it right away!
Awesome job @142ayushkumar

Comment (Demfier) [reply coyote/views.py]: Also, please don't put more than 80 characters in a single line.

**Comment by kaustubhhiware** [comment]
@Demfier the commits should have been squashed before merging. Only one thing was added. So one change per commit. Please revert and only merge after @142ayushkumar has squashed the commits.

**Comment by Demfier** [comment]
@kaustubhhiware Ok, instead of reverting, I am removing those commits from `master`.
A new PR with squashed commits would be better than reverting. What do you say?

**Comment by 142ayushkumar** [comment]
i did this :
 git rebase -i HEAD~5 
but i can't see any difference

**Comment by kaustubhhiware** [comment]
I'll try doing it then in some time.

**Comment by kaustubhhiware** [comment]
No, a new pr will be unnecessary. I'll fix it in some time.

**Comment by kaustubhhiware** [comment]
@Demfier seems that was the only option :confused: 

@142ayushkumar please send your changes from a separate branch again. Sorry for the inconvenience.

Please make sure all changes are in 1 commit only.

**Comment by 142ayushkumar** [comment]
Here: #31

**Pull request merged**